### ISSUE TITLE
Methods as nn.modules

### ISSUE THREAD

**Issue by HEmile** [HEmile/storchastic#23]
If gradient estimation `Method`s are `nn.Module`, we use `def forward(self): self.sample()`, in sample we implement `storch.sample` which refers to an abstract method `sample_tensor` which implements what currently is `sample` in `Method`.

This allows us to store persistent state in `Method` as they are `nn.Module`s of which the state is stored. Allows us to implement baselines, temperature annealing, etc.
